### PR TITLE
Bumps knapsack_pro to v7.1.0

### PR DESCRIPTION
**Pull request by filipefurtad0**
#### What? Why?

- replaces #12424

Context here: https://github.com/openfoodfoundation/openfoodnetwork/pull/12261
This version was superseeded and the context got a but hidden. I'm not sure the issue with webpacker is expected, I've contacted Knapsack support.

Disables Knapsack queue mode and bumps the gem to 7.1.0.

#### What should we test?

- green build

#### Release notes

Changelog Category (reviewers may add a label for the release notes):

- [ ] User facing changes
- [ ] API changes (V0, V1, DFC or Webhook)
- [x] Technical changes only
- [ ] Feature toggled

The title of the pull request will be included in the release notes.

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -203,14 +203,14 @@ jobs:
           # if you use Knapsack Pro Queue Mode you must set below env variable
           # to be able to retry CI build and run previously recorded tests
           # https://github.com/KnapsackPro/knapsack_pro-ruby#knapsack_pro_fixed_queue_split-remember-queue-split-on-retry-ci-node
-          KNAPSACK_PRO_FIXED_QUEUE_SPLIT: true
+          # KNAPSACK_PRO_FIXED_QUEUE_SPLIT: true
```
Comment (rioug) [inline .github/workflows/build.yml:206]: What's the reason behind disabling queue mode ?

File: Gemfile [hunk 2]
```diff
@@ -157,7 +157,7 @@ group :test, :development do
   gem "factory_bot_rails", '6.2.0', require: false
   gem 'fuubar', '~> 2.5.1'
   gem 'json_spec', '~> 1.1.4'
-  gem 'knapsack_pro'
+  gem 'knapsack_pro', '7.0.0'
```
Comment (mkllnk) [inline Gemfile]: We shouldn't restrict the version here. You might do that manually to update to a certain version but you don't need to commit this.

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -203,14 +203,14 @@ jobs:
...
           # RSpec split test files by test examples feature - it's optional
           # https://knapsackpro.com/faq/question/how-to-split-slow-rspec-test-files-by-test-examples-by-individual-it
           #KNAPSACK_PRO_RSPEC_SPLIT_BY_TEST_EXAMPLES: true
           KNAPSACK_PRO_TEST_FILE_PATTERN: "{spec/system/admin/**/*_spec.rb}" 
 
         run: |
-          bundle exec rake knapsack_pro:queue:rspec
+          bundle exec rake knapsack_pro:rspec
```
Comment (mkllnk) [inline .github/workflows/build.yml:213]: I guess that the changes to queue mode don't work well with webpacker but do we know why?

Is it more important to upgrade Knapsack or to have queue mode? We could decide not to upgrade until Knapsack fixed this... or we removed webpacker.

File: Gemfile [hunk 4]
```diff
@@ -157,7 +157,7 @@ group :test, :development do
   gem "factory_bot_rails", '6.2.0', require: false
   gem 'fuubar', '~> 2.5.1'
   gem 'json_spec', '~> 1.1.4'
-  gem 'knapsack_pro'
+  gem 'knapsack_pro', '7.1.0'
```
Comment (mkllnk) [inline Gemfile:160]: This will prevent Dependabot from suggesting any more updates. We shouldn't define a version here unless it's necessary.

**Review by mkllnk — CHANGES_REQUESTED** [review]
Cool that you got it working. I'm not sure that we should go that way though. Is there an issue on the Knapsack repository?

**Comment by filipefurtad0** [comment]
I'd prefer to enable queue mode and get it all to work on this PR, but I don't think I'm there yet. The latest change (explicitly enabling webpacker on each system spec node) fixed the webpacker issue but now we're having:
```

      NoMethodError:
        undefined method `keys' for nil:NilClass
      
              b.slice(*(b.keys - a.keys))
```

pretty much everywhere, like here:

```

  10) full-page cart viewing the cart updating quantities with insufficient stock available shows the quantities saved, not those submitted
      Failure/Error: example.run
      
      NoMethodError:
        undefined method `keys' for nil:NilClass
      
              b.slice(*(b.keys - a.keys))
                         ^^^^^
      # ./spec/system/support/cuprite_setup.rb:39:in `block (2 levels) in <top (required)>'
      # ./spec/base_spec_helper.rb:101:in `block (3 levels) in <main>'
      # ./spec/base_spec_helper.rb:101:in `block (2 levels) in <main>'
```

I'm not sure what to make of this yet...

**Comment by filipefurtad0** [comment]
Closing in favor of #12447 :muscle:

**Pull request closed without merging**